docs/user/troubleshooting.md

```diff
@@ -99,6 +99,237 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+    operator-lifecycle-manager                      True        False         False     17m
+    service-ca                                      True        False         False     30m
+    ```
+
+    To get detailed information on why an individual cluster operator is `Failing` or not yet `Available`, you can check the status of that individual operator, for example `monitoring`:
+
+    ```console
+    $ oc --config=${INSTALL_DIR}/auth/kubeconfig get clusteroperator monitoring -oyaml
+    apiVersion: config.openshift.io/v1
+    kind: ClusterOperator
+    metadata:
+      creationTimestamp: 2019-02-27T22:47:04Z
+      generation: 1
+      name: monitoring
+      resourceVersion: "24677"
+      selfLink: /apis/config.openshift.io/v1/clusteroperators/monitoring
+      uid: 9a6a5ef9-3ae1-11e9-bad4-0a97b6ba9358
+    spec: {}
+    status:
+      conditions:
+      - lastTransitionTime: 2019-02-27T22:49:10Z
+        message: Successfully rolled out the stack.
+        status: "True"
+        type: Available
+      - lastTransitionTime: 2019-02-27T22:49:10Z
+        status: "False"
+        type: Progressing
+      - lastTransitionTime: 2019-02-27T22:49:10Z
+        status: "False"
+        type: Failing
+      extension: null
+      relatedObjects: null
+      version: ""
+    ```
+
+    Again, the cluster operators also publish [conditions][cluster-operator-conditions] like `Failing`, `Available` and `Progressing` that can help user provide information on the current state of the operator:
+
+    ```console
+    $ oc --config=${INSTALL_DIR}/auth/kubeconfig get clusteroperator monitoring -o=jsonpath='{range .status.conditions[*]}{.type}{" "}{.status}{" "}{.message}{"\n"}{end}'
+    Available True Successfully rolled out the stack
+    Progressing False
+    Failing False
+    ```
+
+    Each clusteroperator also publishes the list of objects owned by the cluster operator. To get that information:
+
+    ```console
+    $ oc --config=${INSTALL_DIR}/auth/kubeconfig get clusteroperator kube-apiserver -o=jsonpath='{.status.relatedObjects}'
+    [map[resource:kubeapiservers group:operator.openshift.io name:cluster] map[group: name:openshift-config resource:namespaces] map[group: name:openshift-config-managed resource:namespaces] map[group: name:openshift-kube-apiserver-operator resource:namespaces] map[group: name:openshift-kube-apiserver resource:namespaces]]
+    ```
+
+**NOTE:** Failing to initialize the cluster is usually not a fatal failure in terms of cluster creation as the user can look at the failures from `ClusterOperator` to debug failures for a cluster operator and take actions which will allow `cluster-version-operator` to make progress.
+
+### Installer fails to fetch for console URL
+
+The installer fetches the URL for OpenShift console using the [route][route-object] in `openshift-console` namespace. If the installer fails the fetch the URL for the console:
+
+1. Check if the console router is `Available` or `Failing`
+
+  ```console
```

Comment:
nit: This and the next fenced block are indented by two spaces.  But they need to be indented by four spaces to be included in the `<li>`:

```console
$ curl -s https://github.com/openshift/installer/blob/982563e4e49207082c59a7583ee262a20a05dcad/docs/user/troubleshooting.md | grep -A2 'Check if the console router'
<li>Check if the console router is <code>Available</code> or <code>Failing</code></li>
</ol>
<div class="highlight highlight-text-shell-session"><pre>$ <span class="pl-s1">oc --config=<span class="pl-smi">${INSTALL_DIR}</span>/auth/kubeconfig get clusteroperator console -oyaml</span>
```